### DISCUSSION TITLE
Error in running an example

### DISCUSSION THREAD

**Discussion by thhiep** [abstools/abstools#333] [General]
Hi,
I tried to compile and run example (the monty-hall) and encountered a problem as below - don't know how to fix that.

C:\bin> .\gen\erl\run
escript: exception error: undefined function class_ABS_DC_DeploymentComponent:init_internal/0
in function runtime:start_mod/7 (/Users/rudi/Source/abstools/frontend/build/resources/main/erlang/absmodel/src/runtime.erl, line 147)
in call from runtime:run_mod/8 (/Users/rudi/Source/abstools/frontend/build/resources/main/erlang/absmodel/src/runtime.erl, line 193)
in call from erl_eval:local_func/8 (erl_eval.erl, line 646)
in call from escript:interpret/4 (escript.erl, line 780)
in call from escript:start/1 (escript.erl, line 277)
in call from init:start_em/1 (init.erl, line 1182)
in call from init:do_boot/3 (init.erl, line 871)

**Comment by rudi** [comment]
Hi Hiep Trinh, I'm trying to reproduce this.  Here's what I do (lines beginning with `$` are the commands I input, unmarked lines show the output):

```
$ md5sum monty-hall.abs
f0ba1be9b9772e7cd26e6ff40c3d9c94  monty-hall.abs
$ absc --version
ABS Tool Suite version HEAD
Built from git tree v1.9.3-226-g61b466642
$ absc -e monty-hall.abs
$ gen/erl/run
I always switched, won 664 out of 1000 rounds.
I never switched, won 310 out of 1000 rounds.
$
```

Can you show the output of the same commands on your machine?  Many thanks in advance!

**Reply by thhiep** [reply]
**md5sums C:\bin\absexamples\collaboratory\examples\monty-hall\monty-hall.abs**

MD5sums 1.2 freeware for Win9x/ME/NT/2000/XP+
Copyright (C) 2001-2005 Jem Berkes - http://www.pc-tools.net/
Type C:\bin\md5sums.exe -h for help

[Path] / filename                              MD5 sum
-------------------------------------------------------------------------------
[C:\bin\absexamples\collaboratory\examples\monty-hall\]
monty-hall.abs                                 de2c052a31a5fe512c2c3a399f1c0b78

 **java -jar .\absfrontend.jar --version**
ABS Tool Suite version v1.9.3
Built from git tree v1.9.3

**java -jar .\absfrontend.jar --erlang C:\bin\absexamples\collaboratory\examples\monty-hall\monty-hall.abs -v**
Parsing file C:\bin\absexamples\collaboratory\examples\monty-hall\monty-hall.abs
Typechecking Model...
Generating Erlang code in gen\erl
Compiling erlang files with command: erlc -I gen\erl/absmodel/include -o gen\erl/absmodel/_build/default/lib/absmodel/ebin/ gen\erl\absmodel\src\abs_constructor_info.erl gen\erl\absmodel\src\active_object_task.erl gen\erl\absmodel\src\async_call_task.erl gen\erl\absmodel\src\builtin.erl gen\erl\absmodel\src\class_ABS_DC_CloudProvider.erl gen\erl\absmodel\src\class_ABS_DC_DeploymentComponent.erl gen\erl\absmodel\src\class_ABS_Meta_Runtime.erl gen\erl\absmodel\src\class_MontyHall_Host.erl gen\erl\absmodel\src\class_MontyHall_NonSwitchingContestant.erl gen\erl\absmodel\src\class_MontyHall_SwitchingContestant.erl gen\erl\absmodel\src\clock.erl gen\erl\absmodel\src\cmp.erl gen\erl\absmodel\src\cog.erl gen\erl\absmodel\src\cog_monitor.erl gen\erl\absmodel\src\coverage.erl gen\erl\absmodel\src\dc.erl gen\erl\absmodel\src\dpor.erl gen\erl\absmodel\src\error_transform.erl gen\erl\absmodel\src\future.erl gen\erl\absmodel\src\gc.erl gen\erl\absmodel\src\getopt.erl gen\erl\absmodel\src\init_task.erl gen\erl\absmodel\src\intar.erl gen\erl\absmodel\src\main_app.erl gen\erl\absmodel\src\main_task.erl gen\erl\absmodel\src\mochinum.erl gen\erl\absmodel\src\modelapi_v1.erl gen\erl\absmodel\src\modelapi_v2.erl gen\erl\absmodel\src\m_ABS_DC_funs.erl gen\erl\absmodel\src\m_ABS_JUnit_funs.erl gen\erl\absmodel\src\m_ABS_Meta_funs.erl gen\erl\absmodel\src\m_ABS_Productline_funs.erl gen\erl\absmodel\src\m_ABS_Scheduler_funs.erl gen\erl\absmodel\src\m_ABS_StdLib_funs.erl gen\erl\absmodel\src\m_MontyHall.erl gen\erl\absmodel\src\object.erl gen\erl\absmodel\src\rationals.erl gen\erl\absmodel\src\runtime.erl gen\erl\absmodel\src\task.erl
gen/erl/absmodel/src/abs_constructor_info.erl:3:2: Warning: export_all flag enabled - all functions will be exported
%    3| -compile(export_all).
%     |  ^

c:/bin/gen/erl/absmodel/_build/default/lib/absmodel/ebin/active_object_task.beam: failed to rename c:/bin/gen/erl/absmodel/_build/default/lib/absmodel/ebin/active_object_task.bea# to c:/bin/gen/erl/absmodel/_build/default/lib/absmodel/ebin/active_object_task.beam: permission denied
gen/erl/absmodel/src/active_object_task.erl:3:2: Warning: behaviour task undefined
%    3| -behaviour(task).
%     |  ^

The beam file name was changed, despite the error said so.

**Reply by thhiep** [reply]
I checked out the examples from github: https://github.com/abstools/absexamples

**Reply by rudi** [reply]
Which operating system is this on?  Seems to be Windows, I could try to find a VM to test this on.  Can you tell me how you installed abstools there? I'll try to reproduce the error, so I should be doing the exact same things as you.  It's fine to write in too much detail rather than leave out a step :)

**Reply by thhiep** [reply]
I use Windows, and the pre-built ABS compiler. Thank you.

**Reply by thhiep** [reply]
I switched to Ubuntu 22.04 on a VM box, it worked. So the bug happens on Windows.

**Reply by rudi** [reply]
Good to hear that you got it running!  I can set up a VM with Windows to test, but will be a while until I'll get around to it, so I'm happy that you got it running on Linux.

**Comment by rudi** [comment]
Additionally, could you post the output of `absc -e -v monty-hall.abs` ?  It might be very long, since there's a lot of (harmless) style warnings in our generated Erlang code.
